pandas/core/frame.py

```diff
@@ -3444,7 +3444,7 @@ def _ixs(self, i: int, axis: int = 0):
                 index=self.columns,
                 name=self.index[i],
                 dtype=new_values.dtype,
-            )
+            ).__finalize__(self)
```

Comment:
Apart from the _iLocIndexer, this specific branch (axis=0) is only reached from FrameColumnApply.series_generator
There, the object that is returned by _ixs is only used internally to set its value (in FrameApply.apply_series_generator). It's therefore likely not necessary to call finalize for this?

We could call finalize in the _iLocIndexer. But that would be the only time that finalize is ever called in a _LocationIndexer. Which makes it seem like that's not the right place to do it. 

The else branch of _ixs always calls finalize in _box_col_values. Apparently this is the only place where _box_col_values is used.
We could therefore refactor _ixs so that internal calls that only set values on the returned object never call finalize in either branch. I'm not sure if there is a need for finalize to be called in these cases too, though? I guess not?